feature-libs/pickup-in-store/components/container/cart-pickup-options-container/cart-pickup-options-container.spec.ts

```diff
@@ -0,0 +1,226 @@
+import { CommonModule } from '@angular/common';
+import { ElementRef, ViewContainerRef } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import { I18nTestingModule } from '@spartacus/core';
+import { PreferredStoreService } from '@spartacus/pickup-in-store/core';
+import {
+  IntendedPickupLocationFacade,
+  PickupLocationsSearchFacade,
+} from '@spartacus/pickup-in-store/root';
+import {
+  LaunchDialogService,
+  LAUNCH_CALLER,
+  OutletContextData,
+} from '@spartacus/storefront';
+import { MockPreferredStoreService } from 'feature-libs/pickup-in-store/core/services/preferred-store.service.spec';
+import { BehaviorSubject, Observable, of } from 'rxjs';
+import { MockIntendedPickupLocationService } from '../../../core/facade/intended-pickup-location.service.spec';
+import { CartPickupOptionsContainerComponent } from './cart-pickup-options-container.component';
+import { ActiveCartFacade, Cart } from '@spartacus/cart/base/root';
+
+import createSpy = jasmine.createSpy;
+
+class MockPickupLocationsSearchFacade implements PickupLocationsSearchFacade {
+  startSearch = createSpy();
+  hasSearchStarted = createSpy();
+  isSearchRunning = createSpy();
+  getSearchResults = createSpy().and.returnValue(
+    of([
+      {
+        name: 'preferredStore',
+        stockInfo: {
+          stockLevel: 12,
+        },
+      },
+    ])
+  );
+  clearSearchResults = createSpy();
+  getHideOutOfStock = createSpy();
+  setBrowserLocation = createSpy();
+  toggleHideOutOfStock = createSpy();
+  stockLevelAtStore = createSpy();
+  getStockLevelAtStore = createSpy().and.returnValue(
+    of({ stockLevel: { displayName: 'London School' } })
+  );
+  getStoreDetails = () =>
+    of({
+      displayName: 'London School',
+    });
+  loadStoreDetails = createSpy();
+  setPickupOptionDelivery = createSpy();
+  setPickupOptionInStore = createSpy();
+}
```

Comment:
Can we not reuse the one from `feature-libs/pickup-in-store/core/facade/pickup-locations-search.service.spec.ts`?

Reply:
Or make this a partial implementation, I don't think we use all the methods on it